services/src/main/java/org/fao/geonet/api/records/MetadataApi.java

```diff
@@ -690,13 +694,84 @@ public FeatureResponse getFeatureCatalog(
 
     }
 
+    @io.swagger.v3.oas.annotations.Operation(summary = "Check metadata title not duplicated",
+        description = "Verifies if the metadata title is in use.")
+    @PostMapping(value = "/{metadataUuid:.+}/checkDuplicatedTitle",
+        produces = {MediaType.APPLICATION_JSON_VALUE})
+    @PreAuthorize("hasAuthority('Editor')")
+    @ApiResponses(value = {
+        @ApiResponse(responseCode = "200", description = "Return if the title is duplicated or not."),
+        @ApiResponse(responseCode = "403", description = ApiParams.API_RESPONSE_NOT_ALLOWED_CAN_VIEW)
+    })
+    public ResponseEntity<Boolean> checkMetadataTitleDuplicated(
+        @Parameter(description = API_PARAM_RECORD_UUID,
+            required = true)
+        @PathVariable
+        String metadataUuid,
+        @Parameter(description = "Metadata title to check",
+            required = true)
+        @RequestBody String title,
+        HttpServletRequest request
+    )
+        throws Exception {
+        try {
+            ApiUtils.canViewRecord(metadataUuid, request);
+        } catch (SecurityException e) {
+            Log.debug(API.LOG_MODULE_NAME, e.getMessage(), e);
+            throw new NotAllowedException(ApiParams.API_RESPONSE_NOT_ALLOWED_CAN_VIEW);
+        }
+
+        Set<String> uuidsWithSameTitle = retrieveMetadataUuidsFromTitle(esSearchManager, title, metadataUuid);
+
+        return ResponseEntity.ok(!uuidsWithSameTitle.isEmpty());
+    }
 
     private boolean isIncludedAttributeTable(RelatedResponse.Fcat fcat) {
         return fcat != null
             && fcat.getItem() != null
-            && fcat.getItem().size() > 0
+            && !fcat.getItem().isEmpty()
             && fcat.getItem().get(0).getFeatureType() != null
             && fcat.getItem().get(0).getFeatureType().getAttributeTable() != null
             && fcat.getItem().get(0).getFeatureType().getAttributeTable().getElement() != null;
     }
+
+
+    /**
+     * Retrieves the list of metadata uuids that have the same dataset identifier.
+     *
+     * @param searchMan             ES search manager.
+     * @param metadataTitle         Metadata title to check.
+     * @param metadataUuidToExclude Metadata identifier to exclude from the search.
+     * @return  A list of metadata uuids that have the same metadata title.
+     */
+    private Set<String> retrieveMetadataUuidsFromTitle(EsSearchManager searchMan,
+                                                            String metadataTitle,
+                                                            String metadataUuidToExclude) {
+
+        Set<String> metadataUuids = new HashSet<>();
+
+        String jsonQuery = " {" +
+            "       \"query_string\": {" +
+            "       \"query\": \"+resourceTitleObject.\\\\*.keyword:\\\"%s\\\" -uuid:\\\"%s\\\"\"" +
```

Comment:
Should we also consider really similar titles ? Also maybe this can be combined with https://github.com/geonetwork/core-geonetwork/pull/6822 which suggest keywords from other similar records but is only client side ? It is not exactly the same but has some common ideas ...

Reply:
Could be an improvement, but need to check how to update the query for this and probably report a different message, maybe even returning the titles or the first N titles.

We plan to backport this to `3.12.x` also, this change would not be the same as not sure the similarity can be added with some big impact to that branch.